src/main.cpp

```diff
@@ -1,151 +1,119 @@
-#include <Arduino.h>
+#include <stdio.h>
+#include <string.h>
 
+#include "hx711.h"
+#include "hx711_noblock.pio.h"
+#include "pico/stdlib.h"
 #include "shoot.h"
 #include "utils.h"
 
-void setup()
-{
-    Serial.begin(115200);
+void update_signal(int &key_input) {
+    // l (led)
+    if (key_input == 108) {
+        utils::flash_led(LED_BUILTIN);
+    }
 
-    // Flash LED
-    utils::flash_led(LED_BUILTIN);
+    // b - beep
+    if (key_input == 98) {
+        shoot::throttle_code = 1;
+        shoot::telemetry = 1;
+    }
 
-    // pwm config
-    // Note that PWM needs to be setup first, 
-    // because the dma dreq requires tts::pwm_slice_num
-    tts::pwm_setup();
+    // r - rise
+    if (key_input == 114) {
+        if (shoot::throttle_code >= ZERO_THROTTLE and
+            shoot::throttle_code <= MAX_THROTTLE) {
+            // Check for max throttle
+            if (shoot::throttle_code == MAX_THROTTLE) {
+                printf("Max Throttle reached\n");
+            } else {
+                shoot::throttle_code += 50;
```

Comment:
This is unsafe because the repeating timer could be raised at this point and send an invalid dshot command. This should be implemented as:
`shoot::throttle_code  = min(shoot::throttle_code + 50, MAX_THROTTLE)`